File: backend/src/main/java/com/issuetracker/backend/service/NotificationService.java
package com.issuetracker.backend.service;

import com.issuetracker.backend.domain.entity.User;
import com.issuetracker.backend.dto.response.NotificationDto;

import java.util.List;

public interface NotificationService {
    void createNotification(User user, String message, Long issueId);
    List<NotificationDto> getMyNotifications();
    long getUnreadCount();
    void markAsRead(Long notificationId);
    void markAllAsRead();
}

File: frontend/src/api/notificationService.js
import api from './axiosConfig';

export const notificationService = {
  getMyNotifications: async () => {
    const response = await api.get('/notifications');
    return response.data;
  },
  getUnreadCount: async () => {
    const response = await api.get('/notifications/unread/count');
    return response.data;
  },
  markAsRead: async (id) => {
    const response = await api.put(`/notifications/${id}/read`);
    return response.data;
  },
  markAllAsRead: async () => {
    const response = await api.put('/notifications/read/all');
    return response.data;
  }
};
